remotes/docker/authorizer.go

```diff
@@ -186,15 +186,15 @@ func (a *dockerAuthorizer) AddResponses(ctx context.Context, responses []*http.R
 				return err
 			}
 
-			if username != "" && secret != "" {
-				common := auth.TokenOptions{
-					Username: username,
-					Secret:   secret,
-				}
-
-				a.handlers[host] = newAuthHandler(a.client, a.header, c.Scheme, common)
-				return nil
+			if username == "" || secret == "" {
+				return fmt.Errorf("%w: no basic auth credentials", ErrInvalidAuthorization)
```

Comment:
might as well tell them which part of the credentials was  "missing username" and/or + "missing secret"

Reply:
The way the credential function is used, normally both are empty, both are set (basic auth) or just the secret is set (refresh token). In either case, it would be considered not basic auth credentials. Will leave it up to the user facing functions to validate the credentials or fetch the appropriate secret.